### ISSUE TITLE
should we make isone(::PolynomialElem) and isgen(::PolynomialElem) more correct on zero rings?

### ISSUE THREAD

**Issue by tthsqe12** [Nemocas/AbstractAlgebra.jl#978]
Currently,
```
julia> R, _ = PolynomialRing(ResidueRing(ZZ, 1), "x")
(Univariate Polynomial Ring in x over Residue ring of Integers modulo 1, 0)

julia> isone(one(R))
false

julia> isgen(gen(R))
false
```

Should they be true?

```
-isone(a::PolynomialElem) = length(a) == 1 && isone(coeff(a, 0))
+isone(a::PolynomialElem) = length(a) <= 1 && isone(coeff(a, 0))
 
 function isgen(a::PolynomialElem)
-    return length(a) == 2 && iszero(coeff(a, 0)) && isone(coeff(a, 1))
+    return length(a) <= 2 && iszero(coeff(a, 0)) && isone(coeff(a, 1))
 end
```

**Comment by wbhart** [comment]
Yes, I believe these should be true.

**Comment by fingolfin** [comment]
This has been resolved recently in PR #1910
